Hi

<div style="
    background: linear-gradient(to right, #173B63, #7FB3D5);
    color: white;
    text-align: center;
    padding: 28px;
    border-radius: 18px;
    border: 2px solid #173B63;
    box-shadow: 0 4px 10px rgba(0, 0, 0, 0.35);
    font-size: 34px;
    font-weight: bold;
    letter-spacing: 2px;
    text-shadow: 2px 2px 4px rgba(0, 0, 0, 0.5);
">
    Bank Customer Churn Analysis
</div>


## Project Overview

This project aims to analyze customer data and identify factors
associated with customer churn.

## Objective

The objective of this notebook is to understand the structure,
content, and quality of the raw dataset before data cleaning
and exploratory data analysis.

In [6]:
import pandas as pd
pd.set_option("display.max_columns", None)


## 2. Load Dataset

### 2.1 Check Available Sheets

In [7]:
excel_file = pd.ExcelFile(
    "C:/Users/Redmi/.vscode/Bank_churn/Bank_Churn_Messy.xlsx"
)

print(excel_file.sheet_names)

['Customer_Info', 'Account_Info']


### 2.2 Load the Customer_Info and Account_Info Sheets

In [8]:
# Read Customer_Info sheet
df_customer = pd.read_excel(
    "C:/Users/Redmi/.vscode/Bank_churn/Bank_Churn_Messy.xlsx",
    sheet_name="Customer_Info"
)

# Read Account_Info sheet
df_account = pd.read_excel(
    "C:/Users/Redmi/.vscode/Bank_churn/Bank_Churn_Messy.xlsx",
    sheet_name="Account_Info"
)

print("Customer_Info:")
display(df_customer.head())

print("Account_Info:")
display(df_account.head())

Customer_Info:


,CustomerId,Surname,CreditScore,Geography,Gender,Age,Tenure,EstimatedSalary
0,15634602,Hargrave,619,FRA,Female,42.0,2,€101348.88
1,15647311,Hill,608,Spain,Female,41.0,1,€112542.58
2,15619304,Onio,502,French,Female,42.0,8,€113931.57
3,15701354,Boni,699,FRA,Female,39.0,1,€93826.63
4,15737888,Mitchell,850,Spain,Female,43.0,2,€79084.1


Account_Info:


,CustomerId,Balance,NumOfProducts,HasCrCard,Tenure,IsActiveMember,Exited
0,15634602,€0.0,1,Yes,2,Yes,1
1,15634602,€0.0,1,Yes,2,Yes,1
2,15647311,€83807.86,1,Yes,1,Yes,0
3,15619304,€159660.8,3,No,8,No,1
4,15701354,€0.0,2,No,1,No,0


<div style="
    background: linear-gradient(to right, #173B63, #7FB3D5);
    color: white;
    text-align: center;
    padding: 28px;
    border-radius: 18px;
    border: 2px solid #173B63;
    box-shadow: 0 4px 10px rgba(0, 0, 0, 0.35);
    font-size: 34px;
    font-weight: bold;
    letter-spacing: 2px;
    text-shadow: 2px 2px 4px rgba(0, 0, 0, 0.5);
">
    Dataset Overview
</div>

## 3. Dataset Overview

In [9]:
# Check the number of rows and columns on Customer_Info sheet

print("Number of rows:", df_customer.shape[0])
print("Number of columns:", df_customer.shape[1])

# Display column names of customer sheet

print("Column names:")
print(df_customer.columns.tolist())

# Display dataset information

print("Customer_Info sheet information")
df_customer.info()



# Check the number of rows and columns on Account_Info sheet 

print("\nNumber of rows:", df_account.shape[0])
print("Number of columns:", df_account.shape[1])

# Display column names of account sheet

print("Column names:")
print(df_account.columns.tolist())

# Display dataset information

print("Account_Info sheet information")
df_account.info()

Number of rows: 10001
Number of columns: 8
Column names:
['CustomerId', 'Surname', 'CreditScore', 'Geography', 'Gender', 'Age', 'Tenure', 'EstimatedSalary']
Customer_Info sheet information
<class 'pandas.DataFrame'>
RangeIndex: 10001 entries, 0 to 10000
Data columns (total 8 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   CustomerId       10001 non-null  int64  
 1   Surname          9998 non-null   str    
 2   CreditScore      10001 non-null  int64  
 3   Geography        10001 non-null  str    
 4   Gender           10001 non-null  str    
 5   Age              9998 non-null   float64
 6   Tenure           10001 non-null  int64  
 7   EstimatedSalary  10001 non-null  str    
dtypes: float64(1), int64(3), str(4)
memory usage: 625.2 KB

Number of rows: 10002
Number of columns: 7
Column names:
['CustomerId', 'Balance', 'NumOfProducts', 'HasCrCard', 'Tenure', 'IsActiveMember', 'Exited']
Account_Info sheet information
<class 'pan

#### Based on data types:
The Age column is stored as the float64 data type. This may be due to the presence of missing values. The data type and values should be reviewed during the data-cleaning process.

The EstimatedSalary column is stored as a string/object data type, although salary is expected to be a numerical variable. The column should be examined for formatting issues and converted to an appropriate numerical data type during data cleaning.

<div style="
    background: linear-gradient(to right, #173B63, #7FB3D5);
    color: white;
    text-align: center;
    padding: 28px;
    border-radius: 18px;
    border: 2px solid #173B63;
    box-shadow: 0 4px 10px rgba(0, 0, 0, 0.35);
    font-size: 34px;
    font-weight: bold;
    letter-spacing: 2px;
    text-shadow: 2px 2px 4px rgba(0, 0, 0, 0.5);
">
    Data Quality Check
</div>

## 4. Data Quality Assessment

In [10]:
# Check missing values

missing_values_customer = df_customer.isnull().sum()

print("Missing values of customer sheet:")
print(missing_values_customer)


missing_values_account = df_account.isnull().sum()

print("\nMissing values of account sheet:")
print(missing_values_account)


Missing values of customer sheet:
CustomerId         0
Surname            3
CreditScore        0
Geography          0
Gender             0
Age                3
Tenure             0
EstimatedSalary    0
dtype: int64

Missing values of account sheet:
CustomerId        0
Balance           0
NumOfProducts     0
HasCrCard         0
Tenure            0
IsActiveMember    0
Exited            0
dtype: int64


The Customer_Info sheet contains missing values in two columns: Surname and Age

The Account_Info sheet does not contain any missing values.

Therefore, further data cleaning is required for the Surname and Age columns. The appropriate treatment for these missing values will be determined after examining the affected records and considering the nature and analytical importance of each variable.



In [11]:
# Check duplicated rows in 2 sheet 

duplicate_rows_customer = df_customer.duplicated().sum()

print("Number of duplicated rows in customer_info sheet:", duplicate_rows_customer)

duplicate_rows_account = df_account.duplicated().sum()
print("\nNumber of duplicated rows in account_info sheet:", duplicate_rows_account)

Number of duplicated rows in customer_info sheet: 1

Number of duplicated rows in account_info sheet: 2


In [12]:
# Check duplicated CustomerId values

print(
    "Duplicated CustomerId values in Customer_Info:",
    df_customer["CustomerId"].duplicated().sum()
)

print(
    "Duplicated CustomerId values in Account_Info:",
    df_account["CustomerId"].duplicated().sum()
)

Duplicated CustomerId values in Customer_Info: 1
Duplicated CustomerId values in Account_Info: 2


In [ ]:
# Check the number of unique values in each column in 2 sheet 

unique_values_customer = df_customer.nunique()

print("Number of unique values in customer_info sheet:")
print(unique_values_customer)

# Check the number of the unique values in each column

unique_values_account = df_account.nunique()
print("\nNumber of unique values in account_info sheet:")
print(unique_values_account)

Number of unique values in customer_info sheet:
CustomerId         10000
Surname             2932
CreditScore          460
Geography              5
Gender                 2
Age                   70
Tenure                11
EstimatedSalary     9997
dtype: int64

Number of unique values in account_info sheet:
CustomerId        10000
Balance            6382
NumOfProducts         4
HasCrCard             2
Tenure               11
IsActiveMember        2
Exited                2
dtype: int64


The Customer_Info sheet contains 1 duplicated row. The sheet has 10,001 rows but only 10,000 unique CustomerId values. This result is consistent with the duplicate check and indicates that one customer record is duplicated.

The Account_Info sheet contains 2 duplicated rows. The sheet has 10,002 rows but only 10,000 unique CustomerId values. This result is consistent with the duplicate check and indicates that two customers record are duplicated.

Most categorical variables have a limited number of unique values. For example, Gender has 2 unique values, while HasCrCard, IsActiveMember, and Exited also have 2 unique values. These result are generally valid for binary variables.

The EstimatedSalary column has 9,997 unique values, indicating that most salary values are distinct. However, this column is currently stored as a string and should be checked for formatting issues before being converted to a numerical data type.

In [15]:
# Examine the values in EstimatedSalary

print(
    df_customer["EstimatedSalary"]
    .head(20)
)

0     €101348.88
1     €112542.58
2     €113931.57
3      €93826.63
4       €79084.1
5     €149756.71
6       €10062.8
7     €119346.88
8       €74940.5
9      €71725.73
10     €80181.12
11     €76390.01
12     €26260.98
13    €190857.79
14     €65951.65
15     €64327.26
16      €5097.67
17     €14406.41
18    €158684.81
19     €54724.03
Name: EstimatedSalary, dtype: str


The `EstimatedSalary` column is stored as a string data type because each value contains the euro currency symbol (`€`). As a result, the values cannot be directly converted to a numerical data type. The currency symbol should be removed, and the column should be converted to a numeric data type during the data-cleaning process.
